src/navigation_plugin.cpp

```diff
@@ -2,58 +2,54 @@
 
 #include <ed/entity.h>
 #include <ed/error_context.h>
+#include <ed/convex_hull.h>
+#include <ed/convex_hull_calc.h>
 
 #include <tue/config/reader.h>
 
-#include <iomanip>
+#include <geolib/datatypes.h>
+#include <geolib/Shape.h>
+#include <geolib/CompositeShape.h>
 
-// ----------------------------------------------------------------------------------------------------
-/**
- * @brief transformChull transform all points in chull with pose
- * @param pose pose to be applied to the points in chull
- * @param chull points to be transformed
- */
-void transformChull(const geo::Pose3D& pose, std::vector<geo::Vector3>& chull)
-{
-    for (unsigned int i = 0; i < chull.size(); ++i)
-        chull[i] = pose * chull[i];
-}
+#include <iomanip>
 
 // ----------------------------------------------------------------------------------------------------
 
 /**
  * @brief constructConstraint
- * @param chull Vector with the points of the area. These points should be in the correct order. This means that the points should represent the border of the area.
- * Becuase consecutive point pairs are used to create the contstraint. First point is added to the end, so also the pair 'first-last' is used.
+ * @param ConvexHull with the points of the volume. These points should be in the correct order. This means that the
+ * points should represent the border of the volume. Becuase consecutive point pairs are used to create the contstraint.
+ * First point is added to the end, so also the pair 'first-last' is used.
  * @param constraint string with the contraint
  * @param offset offset to the constraint
  */
-void constructConstraint(std::vector<geo::Vector3>& chull, std::stringstream& constraint, double offset = 0)
+void constructConstraint(ed::ConvexHull& chull, std::stringstream& constraint, double offset = 0)
 {
-    if (chull.size() < 3)
+    if (chull.points.size() < 3)
     {
         std::cout << "Error: Convex hull has to consist of at least three points !!" << std::endl;
         return;
     }
 
-    chull.push_back(chull[0]);
+    std::vector<geo::Vec2f> points = chull.points;
+    points.push_back(points[0]);
 
     // To make sure we don't get e powers in the string
     constraint << std::fixed << std::setprecision(6);
 
     constraint << "(";
 
     double dx,dy,xi,yi,xs,ys,length;
-    for (unsigned int i = 0; i < chull.size()-1; ++i)
+    for (unsigned int i = 0; i < points.size()-1; ++i)
     {
         if (i > 0)
             constraint << " and ";
 
-        xi = chull[i].x;
-        yi = chull[i].y;
+        xi = points[i].x;
+        yi = points[i].y;
 
-        dx = chull[i+1].x - xi;
-        dy = chull[i+1].y - yi;
+        dx = points[i+1].x - xi;
+        dy = points[i+1].y - yi;
```

Comment:
Instead of accessing the `i+1`-th point, do
```` cpp
dx = ((i + 1 < chull.size()) ? chull[i+1].x : chull[0].x) - xi;
dy = ((i + 1 < chull.size()) ? chull[i+1].y : chull[0].y) - yi;
````
thus eliminating the above copy.